### ISSUE TITLE
NaN rotaty_emb weights in exported pretrained models

### ISSUE THREAD

**Issue by ahxxm** [AnswerDotAI/ModernBERT#215]
I used `pretraining_documentation/convert_to_hf.py` to convert pretrain checkpoint pt to HF models, as some model variants don't need to extend context window, but I found that output logits are nan, and `rotaty_emb` is nan in local attention layers

The fix is simple, just change `config.json` content:

* old: `local_rope_theta: -1`
* fixed: `local_rope_theta: 10000.0`, the same as global and as in training

and reload model from dir again, then model outputs totally make sense.

I guess this is because pretrain yamls don't have this value, and FlexBERTConfig [defaults to -1](https://github.com/AnswerDotAI/ModernBERT/blob/8c57a0f01c12c4953ead53d398a36f81a4ba9e38/src/bert_layers/configuration_bert.py#L94), but I'm not quite sure what would be a proper fix.

**Comment by Rijgersberg** [comment]
Thank you, this indeed fixes the same problem I've been having with a checkpoint using a pretrain config!

**Comment by NohTow** [comment]
Good catch!
I wrote the conversion script on top of decayed model, so we had a local_rope_theta set
I guess [this line](https://github.com/AnswerDotAI/ModernBERT/blob/0e9dbe6d36000cfe81114e6937686b4982122a71/convert_to_hf.py#L57C9-L59C47) of the converted should extend the check to -1 value and copy the value from the global layers.

The issue arise from the fact that in the research repo, [we check for equality with -1](https://github.com/AnswerDotAI/ModernBERT/blob/8c57a0f01c12c4953ead53d398a36f81a4ba9e38/src/bert_layers/attention.py#L758) while in the transformers implementation, [they check for None.](https://github.com/huggingface/transformers/blob/5c75087aeee7081025370e10d1f571a11600f1ae/src/transformers/models/modernbert/modeling_modernbert.py#L456)
I guess we could also set it to None, but maybe it's cleaner to set it and not have None in a config?

**Comment by ahxxm** [comment]
I'm ok with either!

**Comment by frammiie** [comment]
Just wanted to say much thanks for the help guys, this was exactly the issue I ran into after pretraining. Much love!

**Comment by NohTow** [comment]
FYI, I fixed it in this [commit](https://github.com/AnswerDotAI/ModernBERT/commit/9fbc9c92867dbe782b660702d83465fceea5d2a0) by copying the global value into the local one when exporting a model that has a local value to -1.
Sorry it took so long, the pre-training branch should be merged soon!
